source/JsMaterialX/CMakeLists.txt

```diff
@@ -0,0 +1,44 @@
+set(OUTPUT_CORE ${CMAKE_CURRENT_SOURCE_DIR}/JsMaterialXCore/)
+set(OUTPUT_FORMAT ${CMAKE_CURRENT_SOURCE_DIR}/JsMaterialXFormat/)
+set(OUTPUT_LIBS ${CMAKE_CURRENT_SOURCE_DIR}/_build/source/)
+
+set(CORE_DEPS ${OUTPUT_CORE}JsTypes.cpp
+    ${OUTPUT_CORE}JsExceptions.cpp
+    ${OUTPUT_CORE}JsUtil.cpp
+    ${OUTPUT_CORE}JsElement.cpp
+    ${OUTPUT_CORE}JsInterface.cpp
+    ${OUTPUT_CORE}JsNode.cpp
+    ${OUTPUT_CORE}JsDefinition.cpp
+    ${OUTPUT_CORE}JsDocument.cpp
+    ${OUTPUT_CORE}JsLook.cpp
+    ${OUTPUT_CORE}JsProperty.cpp
+    ${OUTPUT_CORE}JsGeom.cpp
+    ${OUTPUT_CORE}JsVariant.cpp
+    ${OUTPUT_CORE}JsValue.cpp
+    ${OUTPUT_CORE}JsMaterial.cpp
+    ${OUTPUT_CORE}JsRegisteredVectors.cpp
+    ${OUTPUT_CORE}JsTraversal.cpp
+    ${OUTPUT_CORE}JsObserver.cpp)
+
+set(FORMAT_DEPS ${OUTPUT_FORMAT}JsXmlIo.cpp)
+
+set(LIB_DEPS ${OUTPUT_LIBS}MaterialXCore/libMaterialXCore.a ${OUTPUT_LIBS}MaterialXFormat/libMaterialXFormat.a)
+
+add_executable(MaterialXLib MaterialXLib.cpp)
+
+target_link_libraries(MaterialXLib ${CORE_DEPS} ${FORMAT_DEPS} ${LIB_DEPS})
+
+set(JS_FLAGS "-I${CMAKE_CURRENT_SOURCE_DIR}/../ -std=c++17 -s WASM=1 -s DISABLE_EXCEPTION_CATCHING=0 -s MODULARIZE=1 -s EXPORT_ES6=1 -s USE_ES6_IMPORT_META=0 ")
```

Comment:
When trying to build without using emsmake, emmmake errors show up ( at least on windows )
```
"D:\Work\materialx\MaterialX-adsk-dev\build_public_rel\source\JsMaterialX\MaterialXLib.vcxproj" (default target) (13) ->
(Link target) ->
  LINK : warning LNK4044: unrecognized option '/ID:/Work/materialx/MaterialX-adsk-dev/source/JsMaterialX/../'; ignored [D:\Work\materialx\MaterialX-adsk-dev\build_public_rel\
source\JsMaterialX\MaterialXLib.vcxproj]
  LINK : warning LNK4044: unrecognized option '/std=c++17'; ignored [D:\Work\materialx\MaterialX-adsk-dev\build_public_rel\source\JsMaterialX\MaterialXLib.vcxproj]
  LINK : warning LNK4044: unrecognized option '/s'; ignored [D:\Work\materialx\MaterialX-adsk-dev\build_public_rel\source\JsMaterialX\MaterialXLib.vcxproj]
  LINK : warning LNK4044: unrecognized option '/s'; ignored [D:\Work\materialx\MaterialX-adsk-dev\build_public_rel\source\JsMaterialX\MaterialXLib.vcxproj]
  LINK : warning LNK4044: unrecognized option '/s'; ignored [D:\Work\materialx\MaterialX-adsk-dev\build_public_rel\source\JsMaterialX\MaterialXLib.vcxproj]
  LINK : warning LNK4044: unrecognized option '/s'; ignored [D:\Work\materialx\MaterialX-adsk-dev\build_public_rel\source\JsMaterialX\MaterialXLib.vcxproj]
  LINK : warning LNK4044: unrecognized option '/s'; ignored [D:\Work\materialx\MaterialX-adsk-dev\build_public_rel\source\JsMaterialX\MaterialXLib.vcxproj]
  LINK : warning LNK4044: unrecognized option '/-post-js'; ignored [D:\Work\materialx\MaterialX-adsk-dev\build_public_rel\source\JsMaterialX\MaterialXLib.vcxproj]
  LINK : warning LNK4044: unrecognized option '/-post-js'; ignored [D:\Work\materialx\MaterialX-adsk-dev\build_public_rel\source\JsMaterialX\MaterialXLib.vcxproj]
  LINK : warning LNK4044: unrecognized option '/-post-js'; ignored [D:\Work\materialx\MaterialX-adsk-dev\build_public_rel\source\JsMaterialX\MaterialXLib.vcxproj]
  LINK : warning LNK4044: unrecognized option '/-post-js'; ignored [D:\Work\materialx\MaterialX-adsk-dev\build_public_rel\source\JsMaterialX\MaterialXLib.vcxproj]
  LINK : warning LNK4044: unrecognized option '/-post-js'; ignored [D:\Work\materialx\MaterialX-adsk-dev\build_public_rel\source\JsMaterialX\MaterialXLib.vcxproj]
  LINK : warning LNK4044: unrecognized option '/-post-js'; ignored [D:\Work\materialx\MaterialX-adsk-dev\build_public_rel\source\JsMaterialX\MaterialXLib.vcxproj]
  LINK : warning LNK4044: unrecognized option '/-post-js'; ignored [D:\Work\materialx\MaterialX-adsk-dev\build_public_rel\source\JsMaterialX\MaterialXLib.vcxproj]
  LINK : warning LNK4044: unrecognized option '/-post-js'; ignored [D:\Work\materialx\MaterialX-adsk-dev\build_public_rel\source\JsMaterialX\MaterialXLib.vcxproj]
  LINK : warning LNK4044: unrecognized option '/-post-js'; ignored [D:\Work\materialx\MaterialX-adsk-dev\build_public_rel\source\JsMaterialX\MaterialXLib.vcxproj]
  LINK : warning LNK4044: unrecognized option '/-post-js'; ignored [D:\Work\materialx\MaterialX-adsk-dev\build_public_rel\source\JsMaterialX\MaterialXLib.vcxproj]
  LINK : warning LNK4044: unrecognized option '/-post-js'; ignored [D:\Work\materialx\MaterialX-adsk-dev\build_public_rel\source\JsMaterialX\MaterialXLib.vcxproj]
  LINK : warning LNK4044: unrecognized option '/-post-js'; ignored [D:\Work\materialx\MaterialX-adsk-dev\build_public_rel\source\JsMaterialX\MaterialXLib.vcxproj]
  LINK : warning LNK4044: unrecognized option '/-post-js'; ignored [D:\Work\materialx\MaterialX-adsk-dev\build_public_rel\source\JsMaterialX\MaterialXLib.vcxproj]
  LINK : warning LNK4044: unrecognized option '/-post-js'; ignored [D:\Work\materialx\MaterialX-adsk-dev\build_public_rel\source\JsMaterialX\MaterialXLib.vcxproj]
  LINK : warning LNK4044: unrecognized option '/-post-js'; ignored [D:\Work\materialx\MaterialX-adsk-dev\build_public_rel\source\JsMaterialX\MaterialXLib.vcxproj]
  LINK : warning LNK4044: unrecognized option '/-post-js'; ignored [D:\Work\materialx\MaterialX-adsk-dev\build_public_rel\source\JsMaterialX\MaterialXLib.vcxproj]
  LINK : warning LNK4044: unrecognized option '/-post-js'; ignored [D:\Work\materialx\MaterialX-adsk-dev\build_public_rel\source\JsMaterialX\MaterialXLib.vcxproj]
  LINK : warning LNK4044: unrecognized option '/s'; ignored [D:\Work\materialx\MaterialX-adsk-dev\build_public_rel\source\JsMaterialX\MaterialXLib.vcxproj]
  LINK : warning LNK4044: unrecognized option '/s'; ignored [D:\Work\materialx\MaterialX-adsk-dev\build_public_rel\source\JsMaterialX\MaterialXLib.vcxproj]
  LINK : warning LNK4044: unrecognized option '/Os'; ignored [D:\Work\materialx\MaterialX-adsk-dev\build_public_rel\source\JsMaterialX\MaterialXLib.vcxproj]
  LINK : warning LNK4044: unrecognized option '/-bind'; ignored [D:\Work\materialx\MaterialX-adsk-dev\build_public_rel\source\JsMaterialX\MaterialXLib.vcxproj]


"D:\Work\materialx\MaterialX-adsk-dev\build_public_rel\install.vcxproj" (default target) (1) ->
"D:\Work\materialx\MaterialX-adsk-dev\build_public_rel\ALL_BUILD.vcxproj" (default target) (3) ->
"D:\Work\materialx\MaterialX-adsk-dev\build_public_rel\source\JsMaterialX\MaterialXLib.vcxproj" (default target) (13) ->
(Link target) ->
  LINK : fatal error LNK1104: cannot open file 'WASM=1.obj' [D:\Work\materialx\MaterialX-adsk-dev\build_public_rel\source\JsMaterialX\MaterialXLib.vcxproj]
```

Reply:
I made some changes in 13de8ea for the CMakeLists.txt file. The user does not need to call `emcmake cmake`, instead they can directly call `cmake`. The CMakeLists.txt file is now setting the cmake toolchain to emscripten. The user needs to enable the `MATERIALX_BUILD_JS` option and either specify the `MATERIALX_EMSDK_PATH` option or enable the emsdk environment with the `emsdk_env` script.